pkg/kritis/admission/admission.go

```diff
@@ -57,88 +62,190 @@ var (
 		return
 	}
-	// Next, validate images in the pod against ImageSecurityPolicies in the same namespace
-	isps, err := admissionConfig.fetchImageSecurityPolicies(pod.Namespace)
+	// Validate images in the pod against ImageSecurityPolicies in the same namespace
+	isps, err := admissionConfig.fetchImageSecurityPolicies(ns)
 	if err != nil {
 		glog.Errorf("error getting image security policies: %v", err)
-		w.WriteHeader(http.StatusBadRequest)
+		ar.Response.Allowed = false
+		ar.Response.Result = &metav1.Status{
+			Status:  string(constants.FailureStatus),
+			Message: "error getting image security policies",
+		}
 		return
 	}
 	glog.Infof("Got isps %v", isps)
 	// get the client we will get vulnz from
 	metadataClient, err := admissionConfig.fetchMetadataClient()
 	if err != nil {
 		glog.Errorf("error getting metadata client: %v", err)
-		w.WriteHeader(http.StatusBadRequest)
+		ar.Response.Allowed = false
+		ar.Response.Result = &metav1.Status{
+			Status:  string(constants.FailureStatus),
+			Message: fmt.Sprintf("error getting metadata client %v", err),
+		}
 		return
 	}
 	for _, isp := range isps {
 		for _, image := range images {
 			glog.Infof("Getting vulnz for %s", image)
 			violations, err := admissionConfig.validateImageSecurityPolicy(isp, image, metadataClient)
 			if err != nil {
-				w.WriteHeader(http.StatusBadRequest)
+				ar.Response.Allowed = false
+				ar.Response.Result = &metav1.Status{
+					Status:  string(constants.FailureStatus),
+					Message: fmt.Sprintf("error validating image security policy %v", err),
+				}
 				return
 			}
 			// Check if one of the violations is that the image is not fully qualified
 			for _, v := range violations {
 				if v.Violation == securitypolicy.UnqualifiedImageViolation {
 					glog.Infof("%s is not a fully qualified image", image)
-					returnStatus(constants.FailureStatus, fmt.Sprintf("%s is not a fully qualified image", image), w)
+					ar.Response.Allowed = false
+					ar.Response.Result = &metav1.Status{
+						Status:  string(constants.FailureStatus),
+						Message: fmt.Sprintf("%s is not a fully qualified image", image),
+					}
 					return
 				}
 			}
 			if len(violations) != 0 {
-				defaultViolationStrategy.HandleViolation(image, pod, violations)
-				returnStatus(constants.FailureStatus, fmt.Sprintf("found violations in %s", image), w)
+				defaultViolationStrategy.HandleViolation(image, ns, violations)
+				ar.Response.Allowed = false
+				ar.Response.Result = &metav1.Status{
+					Status:  string(constants.FailureStatus),
+					Message: fmt.Sprintf("found violations in %s", image),
+				}
 				return
 			}
 		}
 	}
-	// TODO: Create Attestations as Occurrences for the given images.
-	// At this point, we can return a success status
-	returnStatus(constants.SuccessStatus, constants.SuccessMessage, w)
+
+}
+
+func reviewPod(pod *v1.Pod, ar *v1beta1.AdmissionReview) {
+	// First, check for a breakglass annotation on the pod
+	if checkBreakglass(pod) {
```

Comment:
You can have break-glass annotation even for deployments.

Reply:
I was going to do this in a separate PR, tracking here:
https://github.com/grafeas/kritis/issues/135